pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +357,25 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    if tz is not None:
```

Comment:
can you add a comment before this section, saying this is where we infer / set fold, and point to the standard library documentation for the same, and any other helpful comments related to this discussion (e.g. imagine a future reader wondering what is going on here).

Reply:
This section is now dead, but there is a similar section when we infer from offset. I've linked to PEP 495 there as the standard library docs don't provide much more information about what fold does than we do in the docstring itself.